jsx/StaticDataTable.js

```diff
@@ -282,22 +285,35 @@ var StaticDataTable = React.createClass({
     // Handle numeric inputs
     if (typeof filterData === 'number') {
       var intData = Number.parseInt(data, 10);
-      return filterData === intData;
+      result = (filterData === intData);
     }
 
     // Handle string inputs
     if (typeof filterData === 'string') {
-      var searchKey = filterData.toLowerCase();
-      var searchString = data.toLowerCase();
+      searchKey = filterData.toLowerCase();
+      searchString = data.toLowerCase();
 
       if (exactMatch) {
-        return searchString === searchKey;
+        result = (searchString === searchKey);
       }
 
-      return (searchString.indexOf(searchKey) > -1);
+      result = (searchString.indexOf(searchKey) > -1);
     }
 
-    return false;
+    // Handle array inputs for multiselects
+    if (typeof filterData === 'object') {
```

Comment:
issue for when you unselect all options, everything gets filtered out

Reply:
after unselecting:
![image](https://user-images.githubusercontent.com/6571449/37614360-fb72dbe8-2b80-11e8-9418-731e3cc3c26f.png)
DataDictIndex state:
![image](https://user-images.githubusercontent.com/6571449/37614448-263bdaa0-2b81-11e8-9d22-cc15e4f79f0d.png)

Reply:
There's an inconsistency with the use of quotes when checking `typeof`. I suggest changing the boolean expression checks on lines `54`, `150`, and `151` to single quotes as well since they are more appropriate for those cirumstances.